### PR TITLE
Add stubs for `_thread.lock`

### PR DESCRIPTION
**Pull request by tomasr8**
Not sure if we still need `LockType` when we have `threading.Lock`.. which is identical :thinking:

### CODE REVIEW

**Review by vinniespina — APPROVED** [review]

File: stdlib/_thread.pyi [hunk 1]
```diff
@@ -16,11 +16,16 @@ class LockType:
...
     def locked(self) -> bool: ...
+    def acquire_lock(self, blocking: bool = True, timeout: float = -1) -> bool: ...
+    def release_lock(self) -> None: ...
+    def locked_lock(self) -> bool: ...
     def __enter__(self) -> bool: ...
     def __exit__(
         self, type: type[BaseException] | None, value: BaseException | None, traceback: TracebackType | None
     ) -> None: ...
 
+lock: TypeAlias = LockType  # noqa: Y042
```
Comment (srittau) [inline stdlib/_thread.pyi:27]: This seems to be new in Python 3.13. It's also a regular alias, not a type alias:

```suggestion
if sys.version_info >= (3, 13):
    lock = LockType
```

File: stdlib/_thread.pyi [hunk 2]
```diff
@@ -16,11 +16,16 @@ class LockType:
     def acquire(self, blocking: bool = True, timeout: float = -1) -> bool: ...
     def release(self) -> None: ...
     def locked(self) -> bool: ...
+    def acquire_lock(self, blocking: bool = True, timeout: float = -1) -> bool: ...
+    def release_lock(self) -> None: ...
+    def locked_lock(self) -> bool: ...
```
Comment (picnixz) [inline stdlib/_thread.pyi:21]: I would group those differently:

```py
def acquire(...)
def acquire_lock(...)
``` 

(The `*_lock` methods are exactly the same as the non-`*_lock` and are legacy methods).

Comment (tomasr8) [reply stdlib/_thread.pyi:21]: I grouped them together same as in the PR that added these methods to `Lock`:

https://github.com/python/typeshed/blob/d2ccc9a723f1a38d234d29ff5519be295e68e25f/stdlib/threading.pyi#L112-L114

Comment (picnixz) [reply stdlib/_thread.pyi:21]: Ah yes, my bad. (Maybe add the "undocumented" comment then?)

**Comment by tomasr8** [comment]
Looks like this has been fixed by another PR.. https://github.com/python/typeshed/pull/12588

**Pull request closed without merging**